tests/infra/node.py

```diff
@@ -427,14 +424,10 @@ def _read_ports(self):
                 with open(rpc_address_file, "r", encoding="utf-8") as f:
                     lines = f.read().splitlines()
                     it = [iter(lines)] * 2
-                for (rpc_host, rpc_port), (_, rpc_interface) in zip(
+                for (_, rpc_port), (_, rpc_interface) in zip(
                     zip(*it), self.host.rpc_interfaces.items()
                 ):
```

Comment:
```suggestion
                for (_, rpc_port), rpc_interface in zip(
                    zip(*it), self.host.rpc_interfaces.values()
                ):
```

I'm pretty sure the `zip(*it)` can also be simplified now, but I'm not sure I even understand what's going on with `it = [iter(lines)] * 2` above

Reply:
I've looked into this and I don't think it can be simplified further. Note that this is only to spin up 1.x nodes too, which output different address files.

Reply:
What does this code actually do though?

`lines` is a list of strings, it then creates a single iterator object over it. It creates a list `it`, with two elements both of them pointing to the same iterator. `zip(*it)` is now creating a zip of an iterator with itself, which is very confusing, and then we loop over the zip, throwing away the first value.

Assuming a naive implementation of zip, my best guess is this iterates over every other element of lines, at indices 1, 3, 5, ... Running this snippet seems to confirm that:
```python
lines = ["a","b","c","d"]
it = [iter(lines)]*2
print([ y for (x, y) in zip(*it) ])
# [ "b", "d" ]
```

In which case you can get the same behaviour with `lines[1::2]`, ie. start at index 1 and increment by 2 each time.
ie.

```python
for rpc_port, rpc_interface in zip(lines[1::2], self.host.rpc_interfaces.values()):
```

(Sorry I realize this is kind of off-topic, I just found the zip to itself very hard to comprehend)